Dart/src/com/jetbrains/lang/dart/analyzer/DartAnalysisServerService.java

```diff
@@ -148,6 +148,14 @@
 
   @NotNull private final EvictingQueue<String> myDebugLog = EvictingQueue.create(DEBUG_LOG_CAPACITY);
 
+  public static String getClientId() {
+    return ApplicationNamesInfo.getInstance().getFullProductName().replaceAll(" ", "-");
```

Comment:
Previously spaces were substituted with an underscore char, now with a hyphen. I do not care, just wanted to make sure it is intentional.